set-properties test doesn't build with GHC 8.6

Tried with --allow-newer so perhaps this first needs an update in QuickCheck. Error:

```
tests/set-properties.hs:257:7: error:
    • No instance for (Control.Monad.Fail.MonadFail Gen)
        arising from a do statement
        with the failable pattern ‘x : xs’
    • In a stmt of a 'do' block: x : xs <- get
      In the expression:
        do x : xs <- get
           put xs
           pure x
      In an equation for ‘step’:
          step
            = do x : xs <- get
                 put xs
                 pure x
    |
257 |       x : xs <- get
    |       ^^^^^^^^^^^^^
```

--- a/tests/set-properties.hs
+++ b/tests/set-properties.hs
@@ -254,9 +254,12 @@ setFromList :: [a] -> Gen (Set a)
 setFromList xs = flip evalStateT xs $ mkArb step (length xs)
   where
     step = do
-      x : xs <- get
-      put xs
-      pure x
+      xxs <- get
+      case xxs of
+        x : xs -> do
+          put xs
+          pure x
+        [] -> error "setFromList"
 
 data TwoSets = TwoSets (Set Int) (Set Int) deriving (Show)
 

